### PR TITLE
Remove unnecessary constructor from DataAnnotationsDefaultErrorMessages

### PR DESCRIPTION
**Pull request by hishamco**
Address @deanmarcussen note https://github.com/OrchardCMS/OrchardCore/pull/5793/files/c658972e60b217c23d3c91cb06ea53454759cc9a#r397679244

### CODE REVIEW

**Comment by hishamco** [comment]
I will assign it to `null` to make the build pass

**Comment by sebastienros** [comment]
This is very dangerous as any code that would somehow reference this class, not even use it, would trigger the static constructor and the properties would trigger a Null Reference Exception.

I would suggest to keep the non-static properties, and make this class abstract (and sealed if that compiles) such that we are sure it's not instantiated.

**Comment by hishamco** [comment]
This safe right now ..

File: src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/DataAnnotationsDefaultErrorMessages.cs [hunk 1]
```diff
@@ -1,118 +1,121 @@
...
+        public string CustomValidationAttributeTypeMustBePublic =>  S["The custom validation type '{0}' must be public."];
 
-        public string CustomValidationAttributeValidationError => S["{0} is not valid."];
+        public string CustomValidationAttributeValidationError =>  S["{0} is not valid."];
 
-        public string CustomValidationAttributeValidatorTypeRequired => S["The CustomValidationAttribute.ValidatorType was not specified."];
+        public string CustomValidationAttributeValidatorTypeRequired =>  S["The CustomValidationAttribute.ValidatorType was not specified."];
 
-        public string DataTypeAttributeEmptyDataTypeString => S["The custom DataType string cannot be null or empty."];
+        public string DataTypeAttributeEmptyDataTypeString =>  S["The custom DataType string cannot be null or empty."];
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/DataAnnotationsDefaultErrorMessages.cs:45]: Extrace space added on every line before `S[]`

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/DataAnnotationsDefaultErrorMessages.cs:45]: Good catch, I will fix it now ..

**Pull request merged**